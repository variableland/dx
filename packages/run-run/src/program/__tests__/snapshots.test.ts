import { afterEach, expect, test } from "bun:test";
import { createTestProgram, execCli, mocked, parseProgram } from "test/helpers";

const { cmd, ctx } = await createTestProgram();
const $ = ctx.shell.$;

const rootCommands = ["help", "--help", "--version", "-v"];

afterEach(() => {
  mocked($).mockClear();
});

for (const cmd of rootCommands) {
  test(`should match command: "${cmd}"`, async () => {
    const { stdout } = await execCli(cmd);

    expect(stdout).toMatchSnapshot();
  });
}

for (const command of cmd.commands) {
  const cmd = command.name();

  test(`should match help message for command "${cmd}"`, async () => {
    const { stdout } = await execCli(`${cmd} --help`);

    expect(stdout).toMatchSnapshot();
  });
}

// these command don't use shell ($) instance
const hardCommands = ["info:pkg", "clean"];

const easyTesteableCommands = cmd.commands.filter((command) => {
  const isHard = hardCommands.some((cmd) => command.name() === cmd);
  return !isHard;
});

for (const command of easyTesteableCommands) {
  const cmd = command.name();

  test(`should match "${cmd}" command`, async () => {
    await parseProgram([cmd]);

    expect($).toHaveBeenCalledTimes(1);
    expect(mocked($).mock.results[0]?.value).toMatchSnapshot();
  });
}
