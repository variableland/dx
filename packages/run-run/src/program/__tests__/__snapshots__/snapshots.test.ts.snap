// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`should match command: "help" 1`] = `
"🦊 R U N - R U N: The CLI toolbox to fullstack common scripts in Variable Land 👊

Usage: rr|run-run [options] <command...>

Options:
  -v, --version                output the version number
  -h, --help                   display help for command

Commands:
  check|test:static [options]  check format and lint issues 🔍
  lint [options]               lint the code 🧹
  fmt|format [options]         format the code 🎨
  tsc|typecheck                check if TypeScript code is well typed 🎨
  clean [options]              delete dirty folders or files such as
                               node_modules, etc 🗑️
  help [command]               display help for command

Acknowledgment:
- kcd-scripts: for main inspiration
  https://github.com/kentcdodds/kcd-scripts

- peruvian news: in honor to Run Run
  https://es.wikipedia.org/wiki/Run_Run
"
`;

exports[`should match command: "--help" 1`] = `
"🦊 R U N - R U N: The CLI toolbox to fullstack common scripts in Variable Land 👊

Usage: rr|run-run [options] <command...>

Options:
  -v, --version                output the version number
  -h, --help                   display help for command

Commands:
  check|test:static [options]  check format and lint issues 🔍
  lint [options]               lint the code 🧹
  fmt|format [options]         format the code 🎨
  tsc|typecheck                check if TypeScript code is well typed 🎨
  clean [options]              delete dirty folders or files such as
                               node_modules, etc 🗑️
  help [command]               display help for command

Acknowledgment:
- kcd-scripts: for main inspiration
  https://github.com/kentcdodds/kcd-scripts

- peruvian news: in honor to Run Run
  https://es.wikipedia.org/wiki/Run_Run
"
`;

exports[`should match command: "--version" 1`] = `
"0.0.0-test
"
`;

exports[`should match command: "-v" 1`] = `
"0.0.0-test
"
`;

exports[`should match help message for command "run" 1`] = `
"Usage: rr run [options] <cmds...>

Arguments:
  cmds        commands to execute in sequence (e.g. 'check tsc')

Options:
  -h, --help  display help for command
"
`;

exports[`should match help message for command "check" 1`] = `
"Usage: rr check|test:static [options]

check format and lint issues 🔍

Options:
  -f, --fix     try to fix issues automatically
  --fix-staged  try to fix staged files only
  -h, --help    display help for command

Under the hood, this command uses the biome CLI to check the code.
"
`;

exports[`should match help message for command "lint" 1`] = `
"Usage: rr lint [options]

lint the code 🧹

Options:
  -c, --check  check if the code is valid (default: true)
  -f, --fix    try to fix all the code
  -h, --help   display help for command

Under the hood, this command uses the biome CLI to lint the code.
"
`;

exports[`should match help message for command "fmt" 1`] = `
"Usage: rr fmt|format [options]

format the code 🎨

Options:
  -c, --check  check if the code is formatted (default: true)
  -f, --fix    format all the code
  -h, --help   display help for command

Under the hood, this command uses the biome CLI to format the code.
"
`;

exports[`should match help message for command "tsc" 1`] = `
"Usage: rr tsc|typecheck [options]

check if TypeScript code is well typed 🎨

Options:
  -h, --help  display help for command

Under the hood, this command uses the TypeScript CLI to check the code.
"
`;

exports[`should match help message for command "clean" 1`] = `
"Usage: rr clean [options]

delete dirty folders or files such as node_modules, etc 🗑️

Options:
  --only-dist  delete 'dist' folders only
  --dry-run    outputs the paths that would be deleted
  -h, --help   display help for command

Under the hood, this command uses the rimraf.js to delete dirty folders or files.
"
`;

exports[`should match help message for command "tools" 1`] = `
"Usage: rr tools [options] [command]

expose the internal tools 🛠️

Options:
  -h, --help      display help for command

Commands:
  biome
  help [command]  display help for command
"
`;
